### PR TITLE
[RM RC1] cleanup and Initial content

### PR DESCRIPTION
**Pull request by rabi-abdel**
Addressing #159
Moving content and diagrams from RM to RC to align with the new scoping of different Work-streams.

### CODE REVIEW

**Comment by rabi-abdel** [comment]
This is just a place holder, I will flag this once content is initialised.

**Comment by mf4716** [comment]
@rabi-abdel - acknowledged as a place holder only- thank you.

**Comment by rabi-abdel** [comment]
**summary of change:**

- restructured RM Ch08
- moved some of the content of RM to RC1 Ch01, Ch03, Ch06, Ch08
- moved the majority of RM Ch08 into Annex (to be more cleaned up)

**Comment by markshostak** [comment]
> 
> 
> **summary of change:**
> 
>     * restructured RM Ch08
> 
>     * moved some of the content of RM to RC1 Ch01, Ch03, Ch06, Ch08
> 
>     * moved the majority of RM Ch08 into Annex (to be more cleaned up)

Given the original purpose in the Annex was to hold Test Cases, and RM will no longer cover that level of detail, I'd suggest we delete RM CH-8-Annex, and move the content to RC. Thoughts?

**Comment by xavier-grall** [comment]
@rabi-abdel 

Given the wide cleanup of RM ch08, I guess it has new objectives which I do not know...

Is there any specific reason to keep performance measurements in chapter 8 ?

I think sections 8.6.4, 8.6.5 and 8.6.6 should be kept in that chapter because they describe quite clearly what would be the method for NFVI hw and/or sw validation (and perhaps also section 8.6.7 for VNF validation method).
However, some minor changes may be needed, especially on figures (Open Compute ?) and VNF ref implementation definition (which is rather a set of pseudo-VNFs required for NFVI validatio tests).

**Comment by rabi-abdel** [comment]
> > **summary of change:**
> > ```
> > * restructured RM Ch08
> > 
> > * moved some of the content of RM to RC1 Ch01, Ch03, Ch06, Ch08
> > 
> > * moved the majority of RM Ch08 into Annex (to be more cleaned up)
> > ```
> 
> Given the original purpose in the Annex was to hold Test Cases, and RM will no longer cover that level of detail, I'd suggest we delete RM CH-8-Annex, and move the content to RC. Thoughts?

Agree. I kept it here as we can decide now what to move and what to keep. ultimately, for Snezka, I propose aiming to completely deleting it.

**Comment by rabi-abdel** [comment]
> @rabi-abdel
> 
> Given the wide cleanup of RM ch08, I guess it has new objectives which I do not know...
> 
> Is there any specific reason to keep performance measurements in chapter 8 ?
> 
> I think sections 8.6.4, 8.6.5 and 8.6.6 should be kept in that chapter because they describe quite clearly what would be the method for NFVI hw and/or sw validation (and perhaps also section 8.6.7 for VNF validation method).
> However, some minor changes may be needed, especially on figures (Open Compute ?) and VNF ref implementation definition (which is rather a set of pseudo-VNFs required for NFVI validatio tests).

@xavier-grall are you suggesting to move performance measurement to RC1?

**Comment by xavier-grall** [comment]
> @xavier-grall are you suggesting to move performance measurement to RC1?
No, it's only a question to better understand the 'new' objectives of chapter 08.

**Comment by rabi-abdel** [comment]
@xavier-grall have a look at let me know your thoughts. I have re-arranged few things.

File: doc/ref_cert/lfn/chapters/chapter08.md [hunk 1]
```diff
@@ -5,16 +5,175 @@
...
+### 8.2.2 Yardstick
+
+### 8.2.3 Bottlenecks
+
+
+### 8.2.4 Test Tools
+1. Shaker:  https://pyshaker.readthedocs.io/en/latest/ (The distributed data-plane testing tool built for OpenStack)
+2. Sonubuoy: https://sonobuoy.io/ It is a diagnostic tool that makes it easier to understand the state of a Kubernetes cluster by running a set of plugins (including Kubernetes conformance tests) in an accessible and non-destructive manner.
+
+### 8.2.5 Scenario Descriptor File (SDF)
```
Comment (gkunz) [inline doc/ref_cert/lfn/chapters/chapter08.md]: The SDF is not a test framework and should be handled in a separate section, i.e. not be subsection of 8.2.

File: doc/ref_cert/lfn/chapters/chapter03.md [hunk 2]
```diff
@@ -5,17 +5,88 @@
...
 <a name="3.2"></a>
-## 3.2 Heading
+## 3.2 Measurement
 
+Test validations will be corroborated, and confirmed, with direct comparison between measured results and documented non-functional requirements (NFRs) for applications, hardware and software configuration settings, and host systems.  Throughput, latency, concurrent connections/threads, are all examples of non-functional requirements which specify criteria which can be used to judge the operation of a system, rather than specific behaviours of the application which are defined by functional requirements.
+
+This section attempts to summarize a categorical list of metrics used for test validations.  **For a complete list of metrics, and requirements, please refer to Reference Model**
+
+<a name="3.2.1"></a>
+### 3.2.1 Performance & Resiliency Measurements
```
Comment (gkunz) [inline doc/ref_cert/lfn/chapters/chapter03.md]: I suggest to separate those chapters as requirements and criteria differ.

File: doc/ref_model/chapters/chapter08-annex.md [hunk 3]
```diff
@@ -139,3 +140,511 @@ Example, or Reference, Functest Status Reporting Artifacts:
...
+<li>Sample VNF, CSAR and Robot Test Cases</li>
+<li>Standardized base Test Plan and Test Case suite</li>
+<li>Standardized certification criteria</li>
+<li>Integration with Dovetail and OVP flow (code, docs, cert criteria, etc.)</li>
+<li>Real, usable documentation</li>
+<li>&ldquo;Just add Water&rdquo; deployment vehicle</li>
+</ol>
+
+<a name="x.x.x"></a>
+### Test Tools
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter08-annex.md]: Suggest removal as those have been moved to chapter 8.

File: doc/ref_model/chapters/chapter08-annex.md [hunk 4]
```diff
@@ -139,3 +140,511 @@ Example, or Reference, Functest Status Reporting Artifacts:
...
+1. Shaker:  https://pyshaker.readthedocs.io/en/latest/ (The distributed data-plane testing tool built for OpenStack)
+2. Sonubuoy: https://sonobuoy.io/ It is a diagnostic tool that makes it easier to understand the state of a Kubernetes cluster by running a set of plugins (including Kubernetes conformance tests) in an accessible and non-destructive manner.
+
+<a name="x.x.x"></a>
+### Scenario Descriptor File (SDF)
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter08-annex.md]: Suggest removal as those have been moved to chapter 8.

File: doc/ref_model/chapters/chapter08-annex.md [hunk 5]
```diff
@@ -139,3 +140,511 @@ Example, or Reference, Functest Status Reporting Artifacts:
...
+
+<a name="x.x.x"></a>
+### RA Verification
+
+**NOT MVP**
+
+Process flow for RA (Infra) Validation and Verfiication
+
+<a name="x.x"></a>
+## Current OVP and CVC Process
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter08-annex.md]: Should be moved to RC 1 as it describes the LFN-based process.

File: doc/ref_model/chapters/chapter08-annex.md [hunk 6]
```diff
@@ -139,3 +140,511 @@ Example, or Reference, Functest Status Reporting Artifacts:
...
+<tr><td>VNF</td>	<td>Virtualised Network Function</td></tr>
+<tr><td>VNF-C</td>	<td>VNF Component (can be hosted on a VM, Container, etc)</td></tr>
+<tr><td>VNFM</td>	<td>VNF Manager</td></tr>
+</table>
+
+- **Verification** will be used to indicate conformance to design requirement specifications.  Activities involved Reviews and Walk-Throughs to ensure the NFVI is delivered per implementation specifications.  
+- **Validations** is used to indicate testing performed to confirm the actual output of a product meets the expected, or desired outcome, or behaviour.  
+
+<a name="x.x.x"></a>
+### Resources
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter08-annex.md]: Suggest removal as those have been moved to chapter 3.

File: doc/ref_cert/lfn/chapters/chapter08.md (hunk 1 above)
Comment (rabi-abdel) [reply doc/ref_cert/lfn/chapters/chapter08.md]: good point. I renamed the title to Relevant Community Projects and Initiatives to be more generic

Comment (rabi-abdel) [reply doc/ref_cert/lfn/chapters/chapter08.md]: will mark as resolved.

File: doc/ref_cert/lfn/chapters/chapter03.md (hunk 2 above)
Comment (rabi-abdel) [reply doc/ref_cert/lfn/chapters/chapter03.md]: do you have a suggestion @gkunz where to place them?

File: doc/ref_model/chapters/chapter08-annex.md (hunk 3 above)
Comment (rabi-abdel) [reply doc/ref_model/chapters/chapter08-annex.md]: Agree. removed. will make as resolved.

File: doc/ref_model/chapters/chapter08-annex.md (hunk 4 above)
Comment (rabi-abdel) [reply doc/ref_model/chapters/chapter08-annex.md]: Agree. Removed. will mark as resolved.

File: doc/ref_model/chapters/chapter08-annex.md (hunk 5 above)
Comment (rabi-abdel) [reply doc/ref_model/chapters/chapter08-annex.md]: Yes, the entire General thoughts section has been commented out in Annex. we can move stuff  into RC1 as needed while RC1 is being developed (the text will not be lost as it is commented out). will resolve

**Comment by rabi-abdel** [comment]
> Re: 8.5 General Thoughts in the RM Ch08 Annex, I'm in agreement with Mark S., that rather than load up the Annex, that this content can move to a similar General Thoughts section at the end of the ref_cert document, in an Appendix until we remove/move content to a final destination

@mf4716 OK. the section has been commented out (i.e. the text is there but is not rendered). this way stuff can be moved from there into RC1 as necessary.

**Comment by rabi-abdel** [comment]
I suggest @xavier-grall to add changes you propose in https://github.com/cntt-n/CNTT/pull/464 into here for simplicity. hope this is OK

**Comment by rabi-abdel** [comment]
I suggest @CsatariGergely to add changes you propose in https://github.com/cntt-n/CNTT/pull/589 into here for simplicity by directly committing into this PR. hope this is OK

File: doc/ref_cert/lfn/chapters/chapter03.md (hunk 2 above)
Comment (gkunz) [reply doc/ref_cert/lfn/chapters/chapter03.md]: nothing fancy: keep it in chapter03 but dedicate a separate heading to the resilience measurements

Comment (rabi-abdel) [reply doc/ref_cert/lfn/chapters/chapter03.md]: resolved in latest commits

**Comment by xavier-grall** [comment]
@rabi-abdel  
 > I suggest @xavier-grall to add changes you propose in #464 into here for simplicity by directly committing into this PR. hope this is OK
Done (1 main and 2 additional minor modifications)
Note that I also separate resiliency measurements in a new section.

Should this chapter include VIM measurements ?
Should this chapter include functional tests (ie neither perf nor resiliency measurements) ? for NFVI ? for VIM ?

**Comment by rabi-abdel** [comment]
can we get few more reviews and approvals please? @gkunz @collivier @markshostak @Gaoweitao

**Comment by rabi-abdel** [comment]
> @rabi-abdel
> 
> > I suggest @xavier-grall to add changes you propose in #464 into here for simplicity by directly committing into this PR. hope this is OK
> 
> Done (1 main and 2 additional minor modifications)
> Note that I also separate resiliency measurements in a new section.
> 
> Should this chapter include VIM measurements ?
> Should this chapter include functional tests (ie neither perf nor resiliency measurements) ? for NFVI ? for VIM ?

VIM is considered part of NFVI, so yes, but may I propose to do that in separate PR once this is merged. 
- Yes functional test can be a sub-section under Measurement Types. it will be good to prepare a PR for it once this is approved.

**Comment by rabiabdel** [comment]
48 hours passed, no objections. 2 reviews and approval received as per process.

**Pull request merged**